week2/The system of the student's score management. v1.0.cpp

```diff
@@ -0,0 +1,160 @@
+#include <stdio.h>
+
+void readscore(int n, long student_number[], int score[]);
+void print_total_and_average(int n, int score[]);
+void rank_by_grade_in_descending_order(int n, long student_number[], int score[]);
+void print_score(int n, long student_number[], int score[]);
+void rank_by_student_number_in_ascending_order(int n, long student_number[], int score[]);
+int search_student_rank_and_grade_by_student_number(int n, long student_number[], int score[]);
+void count_the_percentage_of_the_each_category(int n, long student_number[], int score[]);
+
+int main()
+{
+	long student_number[30];
+	int score[30], n, _switch;
+	printf("1.Input record\n2.Caculate total and average score of course\n3.Sort in descending order by score\n");
+	printf("4.Sort in ascending order by number\n5.search by number\n6.Statistic analysis\n0.Exit\n\n");
+	do {
+		printf("Please enter your choice:");
+		scanf_s("%d", &_switch);
+		switch (_switch)
+		{
+			case 1:
+				printf("Please input the number of the students:");
+				scanf_s("%d", &n);
+				readscore(n, student_number, score);
+				break;
+			case 2:
+				print_total_and_average(n, score);
+				break;
+			case 3:
+				rank_by_grade_in_descending_order(n, student_number, score);
+				printf("Rank by grade in descending order\n");
+				print_score(n, student_number, score);
+				break;
+			case 4:
+				rank_by_student_number_in_ascending_order(n, student_number, score);
+				printf("rank by student number in ascending order\n");
+				print_score(n, student_number, score);
+				break;
+			case 5:
+				search_student_rank_and_grade_by_student_number(n, student_number, score);
+				break;
+			case 6:
+				count_the_percentage_of_the_each_category(n, student_number, score);
+				break;
+			case 0:
+				break;
+		}
+	} while (_switch != 0);
+	return 0;
+}
+
+void readscore(int n, long student_number[], int score[]) {
+	printf("Please input the student's number and his score\n");
+	for (int i = 0; i < n; i++) {

```

Comment:
What will happen if user provides some invalid `n`? For example, when `n >= 30` ?

Reply:
Ok, I will improve it.